lib/main.js

```diff
@@ -11,63 +11,138 @@ import './config.js';
 const {config, util: {BedrockError}} = bedrock;
 const didKeyDriver = driver();
 
-let _keys = null;
+let APPLICATION_IDENTITY;
+const SERVICE_IDENTITIES = new Map();
 
-export const keys = new Proxy({}, {
-  get(target, propKey) {
-    if(!_keys) {
-      throw _invalidStateError();
-    }
-    return _keys[propKey];
-  },
-  getPrototypeOf() {
-    if(!_keys) {
-      throw _invalidStateError();
-    }
-    return Object.getPrototypeOf(_keys);
-  },
-  getOwnPropertyDescriptor(target, prop) {
-    if(!_keys) {
-      throw _invalidStateError();
-    }
-    return Object.getOwnPropertyDescriptor(_keys, prop);
-  },
-  set(target, prop, value) {
-    if(!_keys) {
-      throw _invalidStateError();
-    }
-    _keys[prop] = value;
-    return true;
+export function getAppIdentity() {
+  if(!APPLICATION_IDENTITY) {
+    throw new BedrockError('Application identity not found.', 'NotFoundError');
   }
-});
+  return {...APPLICATION_IDENTITY};
+}
+
+export function getServiceIdentities() {
+  return new Map(SERVICE_IDENTITIES);
+}
 
 bedrock.events.on('bedrock.init', async () => {
-  const {seedBase58} = config['app-key'];
+  const {seeds} = config['app-identity'];
+
+  if(!seeds) {
+    throw new BedrockError('Identity seed(s) not set.', 'InvalidStateError');
+  }
 
-  if(!seedBase58) {
+  if(seeds.app) {
+    await _generateApplicationIdentity({seedDescription: seeds.app});
+    const {id} = seeds.app;
+    const identity = {name: 'app', id};
+    logger.debug('Application identity created.', {identity});
+  } else {
+    const identity = {name: 'app'};
+    logger.debug('Application identity not created.', {identity});
+  }
+
+  if(!seeds.services) {
+    logger.debug('Service identity seeds not found.');
+    return;
+  }
+
+  for(const name of Object.keys(seeds.services)) {
+    const seedDescription = seeds.services[name];
+    // any error thrown here should not be caught, we want startup to fail
+    await _generateServiceIdentity({name, seedDescription});
+    const {id} = seeds.app;
+    logger.debug('Service identity created.', {key: {name, id}});
```

Comment:
I think `{key: {name, id}}` will be confusing. If you want to include the app information, then maybe: `{app: {name: 'app', id}, identity: {name, id: seedDescription.id}}` or something like that.